examples/MagicWeather/app/src/main/java/com/revenuecat/sample/ui/paywall/PaywallAdapter.kt

```diff
@@ -35,19 +38,90 @@ class PaywallAdapter(
         val view = LayoutInflater.from(viewGroup.context)
             .inflate(R.layout.paywall_item, viewGroup, false)
 
-        return PackageViewHolder(view, didChoosePackage)
+        return PackageViewHolder(view, didChoosePaywallItem)
     }
 
     override fun onBindViewHolder(viewHolder: PackageViewHolder, position: Int) {
-        offering?.availablePackages?.get(position)?.let {
+        theItems[position].let {
             val item = it
             viewHolder.item = item
 
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_title).text = item.product.title
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_description).text = item.product.description
-            viewHolder.view.findViewById<TextView>(R.id.paywall_item_price).text = item.product.originalPrice
+            when (it) {
+                is PaywallItem.Title -> {
+                    viewHolder.view.findViewById<TextView>(R.id.paywall_item_options_title).visibility = View.VISIBLE
```

Comment:
at some point, we can clean this up by using databinding (see this article https://medium.com/androiddevelopers/no-more-findviewbyid-457457644885)

unrelated to our paywall-messiness but still a good idea since it's best practice now. maybe we can make a follow-up ticket for that, definitely doesn't need to be this PR